neuralforecast/utils.py

```diff
@@ -439,10 +443,126 @@ def augment_calendar_df(df, freq="H"):
 
     return pd.concat([df, ds_data], axis=1), freq_map[freq]
 
-# %% ../nbs/utils.ipynb 28
+# %% ../nbs/utils.ipynb 29
 def get_indexer_raise_missing(idx: pd.Index, vals: List[str]) -> List[int]:
     idxs = idx.get_indexer(vals)
     missing = [v for i, v in zip(idxs, vals) if i == -1]
     if missing:
         raise ValueError(f"The following values are missing from the index: {missing}")
     return idxs
+
+# %% ../nbs/utils.ipynb 31
+class PredictionIntervals:
+    """Class for storing prediction intervals metadata information."""
+
+    def __init__(
+        self,
+        n_windows: int = 2,
+        method: str = "conformal_distribution",
+    ):
+        """
+        n_windows : int
+            Number of windows to evaluate.
+        method : str, default is conformal_distribution
+            One of the supported methods for the computation of prediction intervals:
+            conformal_error or conformal_distribution
+        """
+        if n_windows < 2:
+            raise ValueError(
+                "You need at least two windows to compute conformal intervals"
+            )
+        allowed_methods = ["conformal_error", "conformal_distribution"]
+        if method not in allowed_methods:
+            raise ValueError(f"method must be one of {allowed_methods}")
+        self.n_windows = n_windows
+        self.method = method
+
+    def __repr__(self):
+        return (
+            f"PredictionIntervals(n_windows={self.n_windows}, method='{self.method}')"
+        )
+
+# %% ../nbs/utils.ipynb 32
+def add_conformal_distribution_intervals(
+    fcst_df: DFType,
+    cs_df: DFType,
+    model_names: List[str],
+    level: List[Union[int, float]],
+    cs_n_windows: int,
+    n_series: int,
+    horizon: int,
+) -> DFType:
+    """
+    Adds conformal intervals to a `fcst_df` based on conformal scores `cs_df`.
+    `level` should be already sorted. This strategy creates forecasts paths
+    based on errors and calculate quantiles using those paths.
+    """
+    fcst_df = ufp.copy_if_pandas(fcst_df, deep=False)
+    alphas = [100 - lv for lv in level]
+    cuts = [alpha / 200 for alpha in reversed(alphas)]
+    cuts.extend(1 - alpha / 200 for alpha in alphas)
+    for model in model_names:
+        scores = cs_df[model].to_numpy().reshape(cs_n_windows, n_series, horizon)
```

Comment:
This works in mlforecast because the outputs are sorted by window and then series, but in neural they're sorted by series first, so this has to be:
```python
scores = cs_df[model].to_numpy().reshape(n_series, cs_n_windows, horizon)
scores = scores.transpose(1, 0, 2)
```

Reply:
Added in fix PR